server/controllers/finance/reports/journal/index.js

```diff
@@ -32,15 +33,64 @@ function postingJournalExport(req, res, next) {
   });
 
   let report;
-
   try {
     report = new ReportManager(REPORT_TEMPLATE, req.session, options);
   } catch (e) {
     return next(e);
   }
 
-  const unposted = Journal.buildTransactionQuery(options, false);
-  const posted = Journal.buildTransactionQuery(options, true);
+  return Journal.journalEntryList(options)
+    .then(rows => report.render({ rows }))
+    .then((result) => {
+      if (result.headers.type === 'xlsx') {
+        res.xls(result.headers.filename, filter(result.report.rows));
+      } else {
+        res.set(result.headers).send(result.report);
+      }
+    })
+    .catch(next)
+    .done();
+}
+
+function dateFormat(date) {
+
+  if (date) {
+    const dateTime = new Date(date);
+    return  moment(dateTime).format('YYYY-MM-DD HH:mm:ss');
+  } else {
```

Comment:
This function could be much simpler:
```js
const DATE_FORMAT = 'YYYY-MM-DD HH:mm:ss';
function dateFormat(date) {
  return date ? moment(date).format(DATE_FORMAT) : date;
}
```
If you wanted to be _really_ fancy, you could use arrow functions, but I don't prefer them here, personally.